### ISSUE TITLE
Button index.d.ts OnPress is missing.

### ISSUE THREAD

**Issue by seppemarotta** [GeekyAnts/NativeBase#2091]
I'm currently using Typescript for my project and I found that In index.d.ts Button OnPress is missing. Is there any roadmap for this ?, maybe i can help with it.
native-base: "^2.6.1"
react: "16.3.1"

**Comment by akhil-geekyants** [comment]
@seppemarotta Checked this. Didn't get this error. 
Button interface extends from `ReactNative.TouchableOpacityProperties`. 
So `Button` will have all those properties of `TouchableOpacity` like `onPress`. 
The onPress you have mentioned in the question has a uppercase `O`(OnPress). try changing that to a lowercase `o`(onPress).

If you are still facing any issue, let us know.

**Comment by SupriyaKalghatgi** [comment]
@seppemarotta Did you check this?

**Comment by seppemarotta** [comment]
I'm sorry , i wasn't able to reply early. I tried to use the onPress property as mencioned by @akhil-geekyants but this msg is triggered `{
	"resource": "/Users/giuseppemarotta/Documents/MH Project/moish-typescript-boilerplate/src/components/LoginScreen.tsx",
	"owner": "typescript",
	"code": "2339",
	"severity": 8,
	"message": "Property 'onPress' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Button> & Readonly<{ children?: ReactNode; }> & Re...'.",
	"source": "ts",
	"startLineNumber": 45,
	"startColumn": 74,
	"endLineNumber": 45,
	"endColumn": 100
}`

The component looks like this : 
`              
  <Button iconLeft block style = {styles.button_style} onPress={this.handlePress} >
                        <Icon name="logo-facebook" />
                        <Text>Log In with Facebook</Text>
   </Button>
`
How can i help you to reproduce this ?

**Comment by akhil-geekyants** [comment]
@seppemarotta checked this with the above button code. Working fine for me. Can you create a test repo for us to check.

**Comment by vinnichase** [comment]
I'm facing exactly the same with onPress on different components and the style property which is really annoying.

Additionally this input:
```jsx
<Input
    onChangeText={this.changeWidth}
    value={this.state.width.toString()}
    keyboardType="numeric"
    style={{ height: 60, width: 0, textAlign: 'center', textAlignVertical: 'center' }}
/>
```
Throws this error:
```
[ts] Type '{ onChangeText: (text: any) => void; value: string; keyboardType: string; style: { height: number...'
has no properties in common with type 'IntrinsicAttributes & IntrinsicClassAttributes<Input> & Readonly<{ children?: ReactNode; }> & Rea...'.
```
For some reason it compiles correctly and runs as expected, same as the other similar issues on here.
Just vscode is complaining.
I was using:
```
CRNA 1.0.0
native-base 2.7.0
react 16.3.1
react-native 0.55.2
```

**Comment by takeo-asai** [comment]
I encountered a similar issue.
In my case, './node_modules/@types/react-navigation/node_modules/@types/react-native' caused a problem.
If you are using @types/react-navigation, I think you should check whether the files are existing.

**Comment by vinnichase** [comment]
Exactly, i restarted code several times with and without `import { createStackNavigator } from 'react-navigation';
` to validate if it causes the warnings and its evident: When react-navigation is imported on vscode startup, some types are interpreted wrong.

I now ended up having a `react-navigation.d.ts` in my project root with `declare module 'react-navigation' {` copied from the latest `@types/react-navigation` but without the node_modules.

**Comment by akhil-geekyants** [comment]
@seppemarotta were you able to fix this. Can you create a test repo for us to test.

**Comment by seppemarotta** [comment]
Sorry for the late reply, I will create a test repo tonight. I'm currently at work so i will be available at 8pm(GMT-5)
I'll let you know once i've created the test repo @akhil-geekyants

**Comment by SupriyaKalghatgi** [comment]
@seppemarotta No updates?

**Comment by SupriyaKalghatgi** [comment]
@seppemarotta Closing this today in case of no response

**Comment by marcelo-schroeder** [comment]
At runtime, this still works, but the problem is that the IDE complains that onPress does not exist. I think this is happening because in native-base/index.d.ts Button extends ReactNative.TouchableOpacityProperties but the correct type definition in react-native/index.d.ts is ReactNative.TouchableOpacityProps.
